solutions/src/main/java/ru/sberbank/school/task02/FxConverter.java

```diff
@@ -0,0 +1,47 @@
+package ru.sberbank.school.task02;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import ru.sberbank.school.task02.util.Volume;
+
+public class FxConverter implements FxConversionService {
+
+  private ExternalQuotesService Quotes;
+
+  FxConverter(ExternalQuotesService q) {
+    Quotes = q;
+  }
+
+  @Override
+  public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal amount) {
+    if (operation == null) {
+      throw new NullPointerException("Operation is null");
+    }
+
+    if (symbol == null) {
+      throw new NullPointerException("Symbol is null");
+    }
+
+    if (amount.compareTo(BigDecimal.ZERO) <= 0) {
+      throw new IllegalArgumentException("Amount less than 0");
+    }
+
+    Quote curQuote = null;
+    BigDecimal curVolume = new BigDecimal(amount.doubleValue());
+
+    for (Quote quote : Quotes.getQuotes(symbol)) {
+      BigDecimal volume = quote.getVolumeSize();
+      if(amount.compareTo(volume)<0){
+          if(curVolume.compareTo(amount)<=0 || volume.compareTo(curVolume) < 0){
+              curQuote = quote;
+              curVolume = volume;
+          }
+      }
+    }
+    return (operation == ClientOperation.BUY) ? curQuote.getOffer() : curQuote.getBid();
```

Comment:
&#39;method def&#39; child has incorrect indentation level 4, expected level should be 8.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1330152586>*